### DISCUSSION TITLE
Auto-refresh temporary credentials

### DISCUSSION THREAD

**Discussion by marwenguesmii** [aws/aws-sdk-js#4411] [Q&A]
Hi, I am setting IAM ROLE for service account and I have a problem. The aws creds that I fetch from the token file are only valid for 12 hours. I tried to refresh them by doing this:
```
if (process.env['NODE_ENV'] == 'production') {
  refreshAwsCredentials();
  // Refresh token one hour before it expires
  setInterval(refreshAwsCredentials, 1000 * 3600 * 11);
}

function refreshAwsCredentials() {
  fromTokenFile({
    // AWS Creds Valid for 12 hours
    durationSeconds: 3600 * 12,
  })().then((credentials) => {
    AWS.config.update({
      accessKeyId: credentials.accessKeyId,
      secretAccessKey: credentials.secretAccessKey,
      sessionToken: credentials.sessionToken,
    });
  });
}
```
But it didn't seem like the right solution . Is there any way for the aws creds to auto-refresh ?

**Comment by RanVaknin** [comment]
Hi @Marwennnne ,

You don't need to manually set the credentials. The Credential provider chain should read the IAM role from the environment variables if they are correctly injected, fetch and refresh the credentials when needed.

so:

```js
const AWS = require('aws-sdk')

const client = new AWS.S3()

// client.getObject(....
```
